kooltalk.1
.rn '' }`
''' $Id: kooltalk.1,v 1.2 2000/04/11 08:27:31 drt Exp $
'''
''' $Log: kooltalk.1,v $
''' Revision 1.2  2000/04/11 08:27:31  drt
''' update to kooltalk 2.1.0a, programming c style
'''
''' Revision 1.1  2000/02/21 12:59:26  drt
''' Initial revision
'''
'''
.de Sh
.br
.if t .Sp
.ne 5
.PP
\fB\\$1\fR
.PP
..
.de Sp
.if t .sp .5v
.if n .sp
..
.de Ip
.br
.ie \\n(.$>=3 .ne \\$3
.el .ne 3
.IP "\\$1" \\$2
..
.de Vb
.ft CW
.nf
.ne \\$1
..
.de Ve
.ft R

.fi
..
'''
'''
'''     Set up \*(-- to give an unbreakable dash;
'''     string Tr holds user defined translation string.
'''     Bell System Logo is used as a dummy character.
'''
.tr \(*W-|\(bv\*(Tr
.ie n \{\
.ds -- \(*W-
.ds PI pi
.if (\n(.H=4u)&(1m=24u) .ds -- \(*W\h'-12u'\(*W\h'-12u'-\" diablo 10 pitch
.if (\n(.H=4u)&(1m=20u) .ds -- \(*W\h'-12u'\(*W\h'-8u'-\" diablo 12 pitch
.ds L" ""
.ds R" ""
'''   \*(M", \*(S", \*(N" and \*(T" are the equivalent of
'''   \*(L" and \*(R", except that they are used on ".xx" lines,
'''   such as .IP and .SH, which do another additional levels of
'''   double-quote interpretation
.ds M" """
.ds S" """
.ds N" """""
.ds T" """""
.ds L' '
.ds R' '
.ds M' '
.ds S' '
.ds N' '
.ds T' '
'br\}
.el\{\
.ds -- \(em\|
.tr \*(Tr
.ds L" ``
.ds R" ''
.ds M" ``
.ds S" ''
.ds N" ``
.ds T" ''
.ds L' `
.ds R' '
.ds M' `
.ds S' '
.ds N' `
.ds T' '
.ds PI \(*p
'br\}
.\"	If the F register is turned on, we'll generate
.\"	index entries out stderr for the following things:
.\"		TH	Title 
.\"		SH	Header
.\"		Sh	Subsection 
.\"		Ip	Item
.\"		X<>	Xref  (embedded
.\"	Of course, you have to process the output yourself
.\"	in some meaninful fashion.
.if \nF \{
.de IX
.tm Index:\\$1\t\\n%\t"\\$2"
..
.nr % 0
.rr F
.\}
.TH KOOLTALK 1 "Chaos Contributed Software" "21/Feb/98"
.IX Title "KOOLTALK 1"
.UC
.IX Name "kooltalk - transform normal text into "elite" kooltalk"
.if n .hy 0
.if n .na
.ds C+ C\v'-.1v'\h'-1p'\s-2+\h'-1p'+\s0\v'.1v'\h'-1p'
.de CQ          \" put $1 in typewriter font
.ft CW
'if n "\c
'if t \\&\\$1\c
'if n \\&\\$1\c
'if n \&"
\\&\\$2 \\$3 \\$4 \\$5 \\$6 \\$7
'.ft R
..
.\" @(#)ms.acc 1.5 88/02/08 SMI; from UCB 4.2
.	\" AM - accent mark definitions
.bd B 3
.	\" fudge factors for nroff and troff
.if n \{\
.	ds #H 0
.	ds #V .8m
.	ds #F .3m
.	ds #[ \f1
.	ds #] \fP
.\}
.if t \{\
.	ds #H ((1u-(\\\\n(.fu%2u))*.13m)
.	ds #V .6m
.	ds #F 0
.	ds #[ \&
.	ds #] \&
.\}
.	\" simple accents for nroff and troff
.if n \{\
.	ds ' \&
.	ds ` \&
.	ds ^ \&
.	ds , \&
.	ds ~ ~
.	ds ? ?
.	ds ! !
.	ds /
.	ds q
.\}
.if t \{\
.	ds ' \\k:\h'-(\\n(.wu*8/10-\*(#H)'\'\h"|\\n:u"
.	ds ` \\k:\h'-(\\n(.wu*8/10-\*(#H)'\`\h'|\\n:u'
.	ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'^\h'|\\n:u'
.	ds , \\k:\h'-(\\n(.wu*8/10)',\h'|\\n:u'
.	ds ~ \\k:\h'-(\\n(.wu-\*(#H-.1m)'~\h'|\\n:u'
.	ds ? \s-2c\h'-\w'c'u*7/10'\u\h'\*(#H'\zi\d\s+2\h'\w'c'u*8/10'
.	ds ! \s-2\(or\s+2\h'-\w'\(or'u'\v'-.8m'.\v'.8m'
.	ds / \\k:\h'-(\\n(.wu*8/10-\*(#H)'\z\(sl\h'|\\n:u'
.	ds q o\h'-\w'o'u*8/10'\s-4\v'.4m'\z\(*i\v'-.4m'\s+4\h'\w'o'u*8/10'
.\}
.	\" troff and (daisy-wheel) nroff accents
.ds : \\k:\h'-(\\n(.wu*8/10-\*(#H+.1m+\*(#F)'\v'-\*(#V'\z.\h'.2m+\*(#F'.\h'|\\n:u'\v'\*(#V'
.ds 8 \h'\*(#H'\(*b\h'-\*(#H'
.ds v \\k:\h'-(\\n(.wu*9/10-\*(#H)'\v'-\*(#V'\*(#[\s-4v\s0\v'\*(#V'\h'|\\n:u'\*(#]
.ds _ \\k:\h'-(\\n(.wu*9/10-\*(#H+(\*(#F*2/3))'\v'-.4m'\z\(hy\v'.4m'\h'|\\n:u'
.ds . \\k:\h'-(\\n(.wu*8/10)'\v'\*(#V*4/10'\z.\v'-\*(#V*4/10'\h'|\\n:u'
.ds 3 \*(#[\v'.2m'\s-2\&3\s0\v'-.2m'\*(#]
.ds o \\k:\h'-(\\n(.wu+\w'\(de'u-\*(#H)/2u'\v'-.3n'\*(#[\z\(de\v'.3n'\h'|\\n:u'\*(#]
.ds d- \h'\*(#H'\(pd\h'-\w'~'u'\v'-.25m'\f2\(hy\fP\v'.25m'\h'-\*(#H'
.ds D- D\\k:\h'-\w'D'u'\v'-.11m'\z\(hy\v'.11m'\h'|\\n:u'
.ds th \*(#[\v'.3m'\s+1I\s-1\v'-.3m'\h'-(\w'I'u*2/3)'\s-1o\s+1\*(#]
.ds Th \*(#[\s+2I\s-2\h'-\w'I'u*3/5'\v'-.3m'o\v'.3m'\*(#]
.ds ae a\h'-(\w'a'u*4/10)'e
.ds Ae A\h'-(\w'A'u*4/10)'E
.ds oe o\h'-(\w'o'u*4/10)'e
.ds Oe O\h'-(\w'O'u*4/10)'E
.	\" corrections for vroff
.if v .ds ~ \\k:\h'-(\\n(.wu*9/10-\*(#H)'\s-2\u~\d\s+2\h'|\\n:u'
.if v .ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'\v'-.4m'^\v'.4m'\h'|\\n:u'
.	\" for low resolution devices (crt and lpr)
.if \n(.H>23 .if \n(.V>19 \
\{\
.	ds : e
.	ds 8 ss
.	ds v \h'-1'\o'\(aa\(ga'
.	ds _ \h'-1'^
.	ds . \h'-1'.
.	ds 3 3
.	ds o a
.	ds d- d\h'-1'\(ga
.	ds D- D\h'-1'\(hy
.	ds th \o'bp'
.	ds Th \o'LP'
.	ds ae ae
.	ds Ae AE
.	ds oe oe
.	ds Oe OE
.\}
.rm #[ #] #H #V #F C
.SH "NAME"
.IX Header "NAME"
kooltalk -- transform normal text into \*(L"elite\*(R" kooltalk
.SH "SYNOPSIS"
.IX Header "SYNOPSIS"
\fIkooltalk\fR [options] [file]
.SH "DESCRIPTION"
.IX Header "DESCRIPTION"
\fBkooltalk\fR will help you to become part of the \*(L"elite\*(R". Ever read
phrases like \*(L"Th1z iS \f(CW@bs0lu\fR+lY k00l, d00d!\*(R" or \*(L"L1k3 Ar1zt0tl3 0r
W1ttg3nzt31n, h3 [H31d3gg3r] l1k3z t0 talk ab0ut h1z w0rdz, and z3ld0m
mak3z an 1nn0vat10n w1th0ut 3xpla1n1ng 1t[...]\*(R" in a README file or an
elite IRC conversation and wondered how this \*(L"3l33t3 hackerz\*(R" were able
to produce a text like that? Ever wished to have that amazing ability?
Well, this is what \fBkooltalk\fR is for (or should it be \*(L"th1z 1z what
\fBkooltalk\fR 1z phor\*(R"?).
.PP
You can pass the name of a file you want to \*(L"koolisize\*(R" on the
commandline. If you don't (or if you give \*(L"\-\*(R" as a filename) \fBkooltalk\fR
will read from stdin and writes to stdout, automagically producing
a translation from \*(L"lame\*(R" to \*(L"k00l\*(R" (these UNIX\-hackerz call this a
\*(L"filter").
.SH "OPTIONS"
.IX Header "OPTIONS"
There are some options available for even more k00lness:
.Ip "\fB\-l\fR" 4
.IX Item "\fB\-l\fR"
Use light koolness (heavy koolness is default). In case you're wondering
what the difference between heavy and light koolness is -- it will all
be revealed in the \f(CWEXAMPLES\fR entry elsewhere in this document . Don't be that impatient, just keep on
reading.
.Ip "\fB\-c\fR" 4
.IX Item "\fB\-c\fR"
Display the copying license. The license is \s-1BSD\s0\-style and very liberal.
Kool program, kool license.
.Ip "\fB\-h\fR" 4
.IX Item "\fB\-h\fR"
Display a short help screen. This isn't much of a help, though. This man
page is n times more useful. But in case you were lame enough to forget
the option to toggle light/heavy koolness (it's \fB\-l\fR, by the way), this
short help could come in quite handy.
.SH "EXAMPLES"
.IX Header "EXAMPLES"
Using \fBkooltalk\fR is easy. To view this man page in a completely k00l way,
you would say:
.Ip "" 4
.IX Item ""
\f(CWman kooltalk | kooltalk | less\fR
.PP
However, it might not necessarily get more understandable that way. \fBkooltalk\fR
asumes that you always want the maximal amount of koolness in your texts and
thus will try to use as many kool alternatives for normal letters as possible.
This is the default \fIheavy koolness mode\fR. Texts translated with heavy
koolness on will not be easily understandable. This is of course more elite,
but not exactly luser-friendly. \fBkooltalk\fR's \fIlight koolness mode\fR (set with
the \fB\-l\fR command line option, remember?) will do less changes to the text,
using only very common koolness characters ("ph\*(R" for \*(L"f\*(R", \*(L"1\*(R" for \*(L"i\*(R" etc).
\fBkooltalk\fR will also try to make intelligent guesses for these changes, i.e.
it will output \*(L"warez\*(R" instead of \*(L"wares\*(R", but will not change \*(L"shut\*(R" to
\*(L"zhut\*(R".
.PP
Being a filter, \fBkooltalk\fR integrates very well with other \s-1UNIX\s0 tools, such as
\fBvi\fR(1). Try the following in \fBvi\fR's command mode for some elite fun:
.Ip "" 4
.IX Item ""
\f(CW:%!kooltalk\fR
.PP
This will transform your boring document to a kool phile. 
.PP
As \fBkooltalk\fR is by definition \*(L"for elite only\*(R", it can be used to protect
your secret philes from the prying eyes of lamers. For extra-elite
cryptographic security, you would use \fBkooltalk\fR together with a similarily
strong encryption software, such as \fBpgp\fR(1). Do it thusly:
.Ip "" 4
.IX Item ""
\f(CWkooltalk < sekret_phile | pgp -fc > supa_sekret_phile\fR
.PP
Use \fBkooltalk \-l\fR for \fIcommercial grade\fR and \fBkooltalk\fR for 
\fImilitary grade\fR encryption.
.SH "BUGS"
.IX Header "BUGS"
I would love to write \*(L"This man page is longer than the program\*(R" (like in 
the \fIcryptdir(1)\fR manpage). Unfortunately, this is is not true. However, the man page is
probabely better written than the source code.  
.SH "SEE ALSO"
.IX Header "SEE ALSO"
\fIvi\fR\|(1), \fIpgp\fR\|(1), \fIcryptdir\fR\|(1), \fIkooltalk\fR\|(1), \fIrecursion\fR\|(3)
.SH "AUTHOR"
.IX Header "AUTHOR"
Jens Ohlig <jo@koeln.ccc.de>

.rn }` ''


kooltalk.c
#include <stdio.h>   
#include <stdlib.h>
#include <unistd.h>
#include <time.h>
#define VERSION "2.1.0"

typedef char stringtype[5];
typedef struct listtype {
  char character;   
  stringtype kool[4];
} listtype;
typedef listtype list[26];
static list koolset;  
static int use_heavy_koolness;
static char filename[256];

static void heavy_koolness()
{
  char c;
  stringtype s;
  short n, i;
  FILE *infile;
  
  if (strlen(filename)>0) {
  	infile = fopen(filename, "r");
	if (infile == NULL) {
		fprintf (stderr, "Couldn't open %s\n", filename);
		exit(1);
	}
}	
else infile = stdin;

while (!feof(infile)) {
	c = fgetc(infile);
	if (c >= 65 && c <= 90)
	c += 32;
	if ((c < 97 || c > 122) && (!feof(infile)))
     	putchar(c);
	if (c < 97 || c > 122)
	continue;
    i = 0;
    while (i <= 25) {  
      if (c == koolset[i].character) {
	n = 1+(int) (3.0*rand()/(RAND_MAX+1.0));
	strcpy(s, koolset[i].kool[n]);
	if (!feof(infile)) {fputs(s, stdout);}   
	i = 26; 
      }
      i++;   
    }
  }
}

static void light_koolness()
{
    char c, d;
    stringtype s;
    FILE *infile;
  
    if (strlen(filename)>0) {
  	infile = fopen(filename, "r");
	if (infile == NULL) {
		fprintf (stderr, "Couldn't open %s\n", filename);
		exit(1);
	}
    }	
    else infile = stdin; 
    while (!feof(infile)) {
    c = fgetc(infile);
    if (feof(infile)) {break;}
    switch (c) {
    case 'i': 
    case 'I': strcpy (s, "1"); fputs(s,stdout); break;
    case 'e': 
    case 'E': strcpy (s, "3"); fputs(s,stdout); break;
    case 'o': 
    case 'O': strcpy (s, "0"); fputs(s,stdout); break;
    case 'f': strcpy (s, "ph"); fputs(s,stdout); break;
    case 'F': strcpy (s, "Ph"); fputs(s,stdout); break;
    case 's': 
    		d = fgetc(infile);
    		if (feof(infile)) {break;} 
    		switch (d) {
			case ' ':
			case '\n':
			case '\t':
			case ',':
			case '.':
			case '?':
			case '!':
			case ';':
			case '-':
			case '_': strcpy (s, "z"); fputs(s,stdout); 
				  putchar(d); break;
			default: putchar(c); putchar(d); break;
		}
		break;
    default: putchar(c);
    } 
  }
}

static void check_params(argc, argv)
int argc;
char *argv[];
{
  char c;
  
  while ((c = getopt(argc, argv, "clh")) !=EOF) {
	switch (c){
        case 'c':
  	printf ("kooltalk %s -- phOR eLi~|~E 0nL`!\n\n", VERSION);
  	printf ("%s", 
"The following copyright statement covers all of kooltalk - either text or
the binary code compiled from the source text.

Copyright (c) 1994-1998 Jens Ohlig <jo@koeln.ccc.de>
All rights reserved.

Redistribution and use in source and binary forms, with or without
modification, are permitted provided that the following conditions
are met:
1. Redistributions of source code must retain the above copyright
   notice, this list of conditions and the following disclaimer.
2. Redistributions in binary form must reproduce the above copyright
   notice, this list of conditions and the following disclaimer in the
   documentation and/or other materials provided with the distribution.
3. All advertising materials mentioning features or use of this software
   must display the following acknowledgement:
	This product includes software developed by Jens Ohlig
	and contributors.
4. Neither the name of the author nor the names of any co-contributors
   may be used to endorse or promote products derived from this software
   without specific prior written permission.

THIS SOFTWARE IS PROVIDED BY THE AUTHOR AND CONTRIBUTORS ``AS IS'' AND
ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
ARE DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR OR CONTRIBUTORS BE LIABLE
FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
SUCH DAMAGE.\n");
	exit(0);
	case 'h': printf ("kooltalk %s -- phOR eLi~|~E 0nL`!\n", VERSION);
		  printf ("%s", "Usage:\n");
		  printf ("%s", "\tkooltalk [options] [file]\n");
		  printf ("%s", "Options:\n");
		  printf ("%s", "\t-l use light koolness ");
		  printf ("%s", "(heavy koolness is default)\n");
	          printf ("%s", "\t-c display the copying license\n");
		  printf ("%s", "\t-h display this help\n"); 
		  exit(0);
	case 'l': use_heavy_koolness = 0; break;
    }
  }
if (argc > optind) strcpy (filename, argv[optind]);
else strcpy (filename, "\0");
if (strchr(filename, 45) != NULL) strcpy (filename, "\0");
}

static void init()
{
  koolset[0].character = 'a';
  koolset[1].character = 'b';
  koolset[2].character = 'c';
  koolset[3].character = 'd';
  koolset[4].character = 'e';
  koolset[5].character = 'f';
  koolset[6].character = 'g';
  koolset[7].character = 'h';
  koolset[8].character = 'i';
  koolset[9].character = 'j';
  koolset[10].character = 'k';
  koolset[11].character = 'l';
  koolset[12].character = 'm';
  koolset[13].character = 'n';
  koolset[14].character = 'o';
  koolset[15].character = 'p';
  koolset[16].character = 'q';
  koolset[17].character = 'r';
  koolset[18].character = 's';
  koolset[19].character = 't';
  koolset[20].character = 'u';
  koolset[21].character = 'v';
  koolset[22].character = 'w';
  koolset[23].character = 'x';
  koolset[24].character = 'y';
  koolset[25].character = 'z';
  strcpy(koolset[0].kool[0], "a");
  strcpy(koolset[0].kool[1], "A");
  strcpy(koolset[0].kool[2], "@");
  strcpy(koolset[0].kool[3], "a");
  strcpy(koolset[1].kool[0], "b");
  strcpy(koolset[1].kool[1], "B");
  strcpy(koolset[1].kool[2], "|3");
  strcpy(koolset[1].kool[3], "b");
  strcpy(koolset[2].kool[0], "c");
  strcpy(koolset[2].kool[1], "C");
  strcpy(koolset[2].kool[2], "(");
  strcpy(koolset[2].kool[3], "[");
  strcpy(koolset[3].kool[0], "d");
  strcpy(koolset[3].kool[1], "D");
  strcpy(koolset[3].kool[2], "c|");
  strcpy(koolset[3].kool[3], "|)");
  strcpy(koolset[4].kool[0], "e");
  strcpy(koolset[4].kool[1], "E");
  strcpy(koolset[4].kool[2], "3");
  strcpy(koolset[4].kool[3], "[-");
  strcpy(koolset[5].kool[0], "f");
  strcpy(koolset[5].kool[1], "F");
  strcpy(koolset[5].kool[2], "/=");
  strcpy(koolset[5].kool[3], "ph");
  strcpy(koolset[6].kool[0], "g");
  strcpy(koolset[6].kool[1], "G");
  strcpy(koolset[6].kool[2], "9");
  strcpy(koolset[6].kool[3], "g");
  strcpy(koolset[7].kool[0], "h");
  strcpy(koolset[7].kool[1], "H");
  strcpy(koolset[7].kool[2], "#");
  strcpy(koolset[7].kool[3], "|-|");
  strcpy(koolset[8].kool[0], "i");
  strcpy(koolset[8].kool[1], "I");
  strcpy(koolset[8].kool[2], "1");
  strcpy(koolset[8].kool[3], "!");
  strcpy(koolset[9].kool[0], "j");
  strcpy(koolset[9].kool[1], "J");
  strcpy(koolset[9].kool[2], "_/");
  strcpy(koolset[9].kool[3], "j");
  strcpy(koolset[10].kool[0], "k");
  strcpy(koolset[10].kool[1], "K");
  strcpy(koolset[10].kool[2], "|<");
  strcpy(koolset[10].kool[3], "k");
  strcpy(koolset[11].kool[0], "l");
  strcpy(koolset[11].kool[1], "L");
  strcpy(koolset[11].kool[2], "|_");
  strcpy(koolset[11].kool[3], "/_");
  strcpy(koolset[12].kool[0], "m");
  strcpy(koolset[12].kool[1], "M");
  strcpy(koolset[12].kool[2], "|\\/|");
  strcpy(koolset[12].kool[3], "m");
  strcpy(koolset[13].kool[0], "n");
  strcpy(koolset[13].kool[1], "N");
  strcpy(koolset[13].kool[2], "|\\|");
  strcpy(koolset[13].kool[3], "/v");
  strcpy(koolset[14].kool[0], "o");
  strcpy(koolset[14].kool[1], "O");
  strcpy(koolset[14].kool[2], "0");
  strcpy(koolset[14].kool[3], "()");
  strcpy(koolset[15].kool[0], "p");
  strcpy(koolset[15].kool[1], "P");
  strcpy(koolset[15].kool[2], "/>");
  strcpy(koolset[15].kool[3], "p");
  strcpy(koolset[16].kool[0], "q");
  strcpy(koolset[16].kool[1], "Q");
  strcpy(koolset[16].kool[2], "(,)");
  strcpy(koolset[16].kool[3], "q");
  strcpy(koolset[17].kool[0], "r");
  strcpy(koolset[17].kool[1], "R");
  strcpy(koolset[17].kool[2], "/~");
  strcpy(koolset[17].kool[3], "/^");
  strcpy(koolset[18].kool[0], "s");
  strcpy(koolset[18].kool[1], "S");
  strcpy(koolset[18].kool[2], "5");
  strcpy(koolset[18].kool[3], "$");
  strcpy(koolset[19].kool[0], "t");
  strcpy(koolset[19].kool[1], "T");
  strcpy(koolset[19].kool[2], "+");
  strcpy(koolset[19].kool[3], "~|~");
  strcpy(koolset[20].kool[0], "u");
  strcpy(koolset[20].kool[1], "U");
  strcpy(koolset[20].kool[2], "(_)");
  strcpy(koolset[20].kool[3], "|_|");
  strcpy(koolset[21].kool[0], "v");
  strcpy(koolset[21].kool[1], "V");
  strcpy(koolset[21].kool[2], "\\/");
  strcpy(koolset[21].kool[3], "v");
  strcpy(koolset[22].kool[0], "w");
  strcpy(koolset[22].kool[1], "W");
  strcpy(koolset[22].kool[2], "|/\\|");
  strcpy(koolset[22].kool[3], "\\/\\/");
  strcpy(koolset[23].kool[0], "x");
  strcpy(koolset[23].kool[1], "X");
  strcpy(koolset[23].kool[2], "><");
  strcpy(koolset[23].kool[3], "x");
  strcpy(koolset[24].kool[0], "y");
  strcpy(koolset[24].kool[1], "Y");
  strcpy(koolset[24].kool[2], "`/");
  strcpy(koolset[24].kool[3], "v/");
  strcpy(koolset[25].kool[0], "z");
  strcpy(koolset[25].kool[1], "Z");
  strcpy(koolset[25].kool[2], "2");
  strcpy(koolset[25].kool[3], "~/_");
}

int main(argc, argv)
int argc;
char *argv[];
{
  time_t mytime;
  mytime = time(0);
  
  srand(mytime);    
  use_heavy_koolness = 1;
  check_params(argc, argv);  
  if (use_heavy_koolness) {
  	init();         
 	heavy_koolness();
  }
  else {
	light_koolness();
  } 
  exit(0);
}
